be/src/olap/memory/typed_column_writer.h

```diff
@@ -0,0 +1,305 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include "olap/memory/column.h"
+#include "olap/memory/column_writer.h"
+#include "olap/memory/typed_column_reader.h"
+
+namespace doris {
+namespace memory {
+
+// Used to hold temporary nullable update cells in ColumnWriter
+template <class T>
+class NullableUpdateType {
+public:
+    bool& isnull() { return _isnull; }
+    T& value() { return _value; }
+
+private:
+    bool _isnull;
```

Comment:
default value

Reply:
I rethought about this, actually, maybe we should avoid initializing those fields for performance reasons, because these fields will be properly assigned in related methods.